### ISSUE TITLE
Absinthe scheme compilation sometimes fails when using import_types multiple modules syntax

### ISSUE THREAD

**Issue by azhi** [absinthe-graphql/absinthe#533]
### Environment

* Elixir version (elixir -v): Elixir 1.6.2 (compiled with OTP 20)
* Absinthe version (mix deps | grep absinthe): * absinthe 1.5.0-dev (../../absinthe) (mix) (current HEAD, a0e2bbf)
* Client Framework and version (Relay, Apollo, etc): -

### Expected behavior

Absinthe schema is always compiled successfully

### Actual behavior

Sometimes compilation fails with

```
== Compilation error in file lib/schema.ex ==
** (ArgumentError) module Elixir.SubSchema is not available
    lib/absinthe/schema/notation.ex:1266: anonymous fn/5 in Absinthe.Schema.Notation.do_import_types/2
    (elixir) lib/enum.ex:1899: Enum."-reduce/3-lists^foldl/2-0-"/3
    lib/absinthe/schema/notation.ex:1260: Absinthe.Schema.Notation.do_import_types/2
    expanding macro: Absinthe.Schema.Notation.import_types/1
    lib/schema.ex:5: App.Schema (module)
```

### Relevant Schema/Middleware Code

lib/schema.ex:
```elixir
defmodule App.Schema do
  use Absinthe.Schema

  import_types(App.A)
  import_types(App.{B, C})

  query do
    import_fields :a_queries
    import_fields :b_queries
    import_fields :c_queries
  end
end
```

lib/a.ex:
```elixir
defmodule App.A do
  use Absinthe.Schema.Notation
  object :a_queries do
    field :a, :integer do
      resolve fn _, _ -> {:ok, 1} end
    end
  end
end
```

lib/b.ex:
```elixir
defmodule App.B do
  use Absinthe.Schema.Notation
  object :b_queries do
    field :b, :integer do
      resolve fn _, _ -> {:ok, 2} end
    end
  end
end
```

lib/c.ex:
```elixir
defmodule App.C do
  use Absinthe.Schema.Notation
  object :c_queries do
    field :c, :integer do
      resolve fn _, _ -> {:ok, 3} end
    end
  end
end
```

The easiest way to reproduce is to run compilation in a cycle:
`while [[ $? == 0 ]]; do mix do clean, compile; done`.
On my PC it fails on 1-8 attempt.

**Comment by azhi** [comment]
Hey @benwilson512.
It seems this was broken in master, I can reproduce the error using steps in issue description.

It was broken by 01be95b540c3a2a4419fd2e8b117ef1f9eb8a569:
Before - https://github.com/absinthe-graphql/absinthe/commit/01be95b540c3a2a4419fd2e8b117ef1f9eb8a569?diff=split#diff-a6bbc7540d5fe20d470c41d27b93489dL1263
After - https://github.com/absinthe-graphql/absinthe/commit/01be95b540c3a2a4419fd2e8b117ef1f9eb8a569?diff=split#diff-a6bbc7540d5fe20d470c41d27b93489dR1423.

Fix in #536 was overwritten as well, so #535 should be reproducible as well.

I didn't dig into how SDL support is implemented and whether these fixes somehow hurt SDL support (huge commit diff scared me away :)). If you think it was just a bad git merge with old code, I can create a PR that restores these fixes.

**Comment by tcoopman** [comment]
Is there any reason these fixes are not merged in `v1.5`?
